plugins/snapd/fu-snapd-snap.c

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 Maciej Borzecki <maciej.borzecki@canonical.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "fu-snapd-snap.h"
+
+/**
+ * fu_snapd_is_in_snap:
+ *
+ * Check whether the current process is running inside a snap.
+ *
+ * Returns: TRUE if current process is running inside a snap.
+ **/
+gboolean
+fu_snapd_is_in_snap(void)
```

Comment:
Do you think this might be useful to other plugins in the future as well?  I would think it will come up again.  If so; this is probably better to put in libfwupdplugin.